src/Translation.jl

```diff
@@ -0,0 +1,338 @@
+
+```jldoctest translation
+julia> tr = (X+v)+v
+Translation{Float64,Array{Float64,1},BallInf{Float64}}(BallInf{Float64}([2.0, 2.0, 2.0], 1.0), [2.0, 0.0, 0.0])
+
+julia> tr.v
+3-element Array{Float64,1}:
+ 2.0
+ 0.0
+ 0.0
+```
+
+The dimension of a translation is obtained with the `dim` function:
+
+```jldoctest translation
+julia> dim(tr)
+3
+```
+
+For the support vector (resp. support function) along vector `d`, use `σ` and
+`ρ` respectively:
+
+```jldoctest translation
+julia> σ([1.0, 0.0, 0.0], tr)
+3-element Array{Float64,1}:
+ 5.0
+ 3.0
+ 3.0
+
+julia> ρ([1.0, 0.0, 0.0], tr)
+5.0
+```
+See the docstring of each of these functions for details.
+
+The `an_element` function is useful to obtain an element of a translation:
+
+```jldoctest translation
+julia> e = an_element(tr)
+3-element Array{Float64,1}:
+ 4.0
+ 2.0
+ 2.0
+ ```
+
+The lazy linear map of a translation is again a translation, since the following
+simplification rule applies: ``M * (X⊕v) = (M*X) ⊕ (M*v)``:
+
+```jldoctest translation
+julia> Q = Matrix(2.0I, 3, 3) * tr;
+
+julia> Q isa Translation && Q.v == 2 * tr.v
+true
+```
+
+Use the `isempty` method to query if the translation is empty; it falls back
+to the `isempty` method of the wrapped set:
+
+```jldoctest translation
+julia> isempty(tr)
+false
+```
+
+The list of constraints of the translation of a polyhedron (in general, a set
+whose `constraints_list` is available) can be computed from a lazy translation:
+
+```jldoctest translation
+julia> constraints_list(tr)
+6-element Array{HalfSpace{Float64},1}:
+ HalfSpace{Float64}([1.0, 0.0, 0.0], 5.0)
+ HalfSpace{Float64}([0.0, 1.0, 0.0], 3.0)
+ HalfSpace{Float64}([0.0, 0.0, 1.0], 3.0)
+ HalfSpace{Float64}([-1.0, -0.0, -0.0], -3.0)
+ HalfSpace{Float64}([-0.0, -1.0, -0.0], -1.0)
+ HalfSpace{Float64}([-0.0, -0.0, -1.0], -1.0)
+```
+"""
+struct Translation{N<:Real, VN<:AbstractVector{N}, S<:LazySet{N}} <: LazySet{N}
+    X::S
+    v::VN
+
+    # default constructor with dimension check
+    function Translation(X::S, v::VN) where {N, VN<:AbstractVector{N}, S<:LazySet{N}}
+        @assert dim(X) == length(v) "cannot create a translation of a set of dimension $(dim(X)) " *
+                                    " along a vector of length $(length(v))" 
+        return new{N, VN, S}(X, v)
+    end
+end
+
+@static if VERSION < v"0.7-"
+    @eval begin
+        # convenience constructor without type parameter
+        Translation(X::S, v::VN) where {N<:Real, VN<:AbstractVector{N}, S<:LazySet{N}} =
+            Translation{N, VN, S}(X, v)
+    end
+end
+
+# constructor from a Translatio:: perform the translation immediately
+Translation(tr::Translation{N}, v::AbstractVector{N}) where {N<:Real} =
+    Translation(tr.X, tr.v + v)
+
+"""
+    X + v
+
+Convenience constructor for a translation.
+
+### Input
+
+- `X` -- convex set
+- `v` -- vector
+
+### Output
+
+The symbolic translation of ``X`` along vector ``v``.
+"""
++(X::LazySet, v::AbstractVector) = Translation(X, v)
```

Comment:
I think you should/have to add `+` and `⊕` to the docs.

Reply:
For the record, there are several problems with the docs (see [here](https://travis-ci.org/JuliaReach/LazySets.jl/jobs/499554056#L804)).